### ISSUE TITLE
Training code release 

### ISSUE THREAD

**Issue by Randy1435** [neonbjb/tortoise-tts#177]
Hey, so I've read that the training code for this repo won't be released, would you be able to provide me with the reasoning as to why? I just don't seem to know why it won't be released. Eventually someone is just going to make their own training scripts for this and publish it on somewhere like Google colab or something.

Please give me concrete reasoning!
Thanks!

**Comment by neonbjb** [comment]
Hi There,
This model is extremely potent when fine-tuned. It can be easily used to make deep-fake voiceovers of people. I'm sure you've heard the Joe Rogan and Steve Jobs podcast - that was Tortoise. Withholding the VQVAE which is required to perform that fine-tuning is my way of de-weaponizing it, to an extent.

> Eventually someone is just going to make their own training scripts for this and publish it on somewhere like Google colab or something.
You're right. It won't be me that releases it, though. Even if this wasn't a question of morals (which it absolutely is), I do not want to be legally responsible for the misuse of this model.

**Comment by Randy1435** [comment]
> You're right. It won't be me that releases it, though. Even if this wasn't a question of morals (which it absolutely is), I do not want to be legally responsible for the misuse of this model.

Even though you won't release the model's training scripts, would you be able to tell me how long it would take to train and how much data you would need to train?

What have you successful results or even believable ones.

**Comment by rebotnix** [comment]
@Randy1435 , @neonbjb , we work in the field of e-learning in schools and we want to use this technology to distribute knowledge for education. So there are many areas where this technology could be useful and economically viable. Maybe there is an official way with the author to get a license with the licensing terms how to use it internally? Alternatively, we can consider whether the community could manage to develop the project further. What do you think?

**Comment by Randy1435** [comment]
> @Randy1435 , @neonbjb , we work in the field of e-learning in schools and we want to use this technology to distribute knowledge for education. So there are many areas where this technology could be useful and economically viable. Maybe there is an official way with the author to get a license with the licensing terms how to use it internally? Alternatively, we can consider whether the community could manage to develop the project further. What do you think?

This is kinda off topic honestly, it would've been a better idea to open another issue, I don't work with or for neonbjb, I was just curious about the questions I asked.

**Comment by rebotnix** [comment]
Right, i created a discord server with some channels, maybe that is an better place instead of using github issues to talk about this. Join if you like: https://discord.gg/6wcmd2Kq

**Comment by midihex** [comment]
With absolute respect, its a great sounding TTS but it's a horror show spectrally with many tells and wouldn't fool anyone with any sort of credibility. Your choice though.

**Comment by plowsjordan** [comment]
@rebotnix I think you have the right idea this needs to be a community effort and when built together could unlock massive possibilities beyond the current SOTA.

**Comment by plowsjordan** [comment]
@midihex What do you think the right move here is?

**Comment by midihex** [comment]
> @midihex What do you think the right move here is? Would love to hear your thoughts

well there's now real-time voice cloners within the past few months that can surpass this easily. Not to mention closed source internal models, which if it is being weaponised seriously would be the ones used, not this.  I said what I said because whilst this is a great TTS it's not perfect by any means, even the fined tuned demos. I get why the author has a fear of releasing his vae but I think its unjust and paranoid really and as for morally? - it already has working models of some celebs, so....

**Comment by plowsjordan** [comment]
@midihex I think you have the right idea here and it needs to be flushed out more, I am putting together a community effort to continue where this work left off. 

You would be a huge value add, would this interest you? If so please join the discord https://discord.gg/aksp6W3t2c channel-name:light-tts

**Comment by chrisbward** [comment]
https://voice.ai surpasses this and it's real-time, so I think it's about time now

**Comment by chavinlo** [comment]
> @midihex I think you have the right idea here and it needs to be flushed out more, I am putting together a community effort to continue where this work left off.
> 
> You would be a huge value add, would this interest you? If so please join the discord https://discord.gg/aksp6W3t2c channel-name:light-tts

Just checked that discord server and now it's named "Light ai" and it's literally a AI as a service project? Really misleading imo.

**Comment by chavinlo** [comment]
Furthermore, one user already got the training code working. I will try to contact him and if he denies I will try it my self

**Comment by neonbjb** [comment]
It's really not that hard. Get some GPUs, read my write up and the referenced papers and do it! If I can do it, anyone can! :)

**Comment by nathanmacauu** [comment]
> 

Hi mate, any good news on that please?
Do we have a community here, I think that quite some users.
Would you mind to add me on Discord please? Thanks.
BP#8658

**Comment by rikabi89** [comment]
Hey considering ultra realistic models are now available in the form of elevenlabs.io and others. Are you going to reconsider releasing the training code?

**Comment by Inthemorningsir** [comment]
Dear Mr. Betker,

are you willing to go into further detail regarding your initial concerns and how, if they have changed, are to be understood today, given the alternatives available? The technology is in the _wrong_ hands for much longer than the public. Many people would like to hear your arguments and know if you have set a condition for yourself under what circumstances you would consider publishing.

Kind regards

**Comment by I-Have-No-Idea-What-IAmDoing** [comment]
He did talked about a lot of his recent thoughts here: https://github.com/neonbjb/tortoise-tts/discussions/292

**Comment by neonbjb** [comment]
I have already "published" everything you need to reproduce Tortoise, including all of the code. I cannot publish the training dataset for legal reasons.

**Comment by neonbjb** [comment]
I believe what you actually want from me is to "build a tool that allows us to easily fine-tune Tortoise to create voice clones of real people". I won't be building that. I'm sorry.

**Comment by TortoiseFanboy** [comment]
> I believe what you actually want from me is to "build a tool that allows us to easily fine-tune Tortoise to create voice clones of real people". I won't be building that. I'm sorry.

That's kinda what everyone wants, it'd be a great thing for everyone

**Comment by TortoiseFanboy** [comment]
I would pay for it honestly

**Comment by adhikjoshi** [comment]
Now Bark and other models have came, Imo, its save to release small portion of code. @neonbjb
